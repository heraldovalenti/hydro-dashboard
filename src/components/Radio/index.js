import React from "react";
import { withStyles } from "@material-ui/core/styles";
import { grey } from "@material-ui/core/colors";
import Radio from "@material-ui/core/Radio";

const StyledRadio = withStyles({
  root: {
    color: grey[400],
    "&$checked": {
      color: grey[800]
    }
  },
  checked: {}
})(props => <Radio color="default" {...props} />);

export default StyledRadio;
