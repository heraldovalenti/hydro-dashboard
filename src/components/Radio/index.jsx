import React from 'react';
import { withStyles } from '@mui/styles';
import { grey } from '@mui/material/colors';
import Radio from '@mui/material/Radio';

const StyledRadio = withStyles({
  root: {
    color: grey[400],
    '&$checked': {
      color: grey[800],
    },
  },
  checked: {},
})((props) => <Radio color="default" {...props} />);

export default StyledRadio;
