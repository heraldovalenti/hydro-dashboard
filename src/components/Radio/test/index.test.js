import React from 'react';
import Radio from '../';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('<Radio/>', () => {
    it('renders no items', () => {
        const renderer = new ShallowRenderer();
        renderer.render(<Radio items={[]}/>);
        const tree = renderer.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });

    it('renders several items', () => {
        const items = [{value: 1, label: 'one'}, {value: 2, label: 'two'}];
        const renderer = new ShallowRenderer();
        renderer.render(<Radio items={items}/>);
        const tree = renderer.getRenderOutput();
        expect(tree).toMatchSnapshot();
    });
});