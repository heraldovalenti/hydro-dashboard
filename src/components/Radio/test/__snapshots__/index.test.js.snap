// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio/> renders no items 1`] = `
<Unknown
  classes={
    Object {
      "checked": "Component-checked-2",
      "root": "Component-root-1",
    }
  }
  items={Array []}
/>
`;

exports[`<Radio/> renders several items 1`] = `
<Unknown
  classes={
    Object {
      "checked": "Component-checked-2",
      "root": "Component-root-1",
    }
  }
  items={
    Array [
      Object {
        "label": "one",
        "value": 1,
      },
      Object {
        "label": "two",
        "value": 2,
      },
    ]
  }
/>
`;
